### PR TITLE
osm 0.1.0 (new formula)

### PR DESCRIPTION
**Pull request by tharun208**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by dawidd6** [comment]
Install method could be reduced to:
```ruby
  def install
    system "make", "build-osm"
    bin.install bin/"osm"
  end
```

**Comment by tharun208** [comment]
> Install method could be reduced to:
> 
> ```ruby
>   def install
>     system "make", "build-osm"
>     bin.install bin/"osm"
>   end
> ```

I tried this and the above didn't work. Can we go with the current install method?

**Comment by dawidd6** [comment]
Care to explain what was wrong?

**Comment by tharun208** [comment]
> Care to explain what was wrong?

we need to set the `srcpath`, so that `make build` will add binary inside `src` folder.

**Comment by dawidd6** [comment]
Sorry but I don't understand. Would be great if you provided some build logs.

**Comment by tharun208** [comment]
> Sorry but I don't understand. Would be great if you provided some build logs.

` Errno::ENOENT: No such file or directory - /usr/local/Cellar/osm/0.1.0/bin/osm`

File: Formula/osm.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  def install
+    srcpath = buildpath/"src/openservicemesh/osm"
+    srcpath.install buildpath.children
+
+    cd srcpath do
+      system "make", "build-osm"
+      prefix.install_metafiles
+      bin.install "bin/osm"
+    end
```
Comment (SeekingMeaning) [inline Formula/osm.rb:18]:
```suggestion
    system "make", "build-osm"
    bin.install "bin/osm"
```

**Comment by SeekingMeaning** [comment]
@dawidd6 probably made a typo and meant `"bin/osm"` instead of `bin/"osm"`

**Review by SeekingMeaning — DISMISSED** [review]

**Comment by dawidd6** [comment]
Please name your commit as PR title.

**Review by chenrui333 — APPROVED** [review]

**Pull request closed without merging**